.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,100 @@
+name: integration-tests
+on:
+  push:
+    branches: [ "518-implement-semi-automatic-integration-tests" ]
+  pull_request:
+    branches: [ "518-implement-semi-automatic-integration-tests" ]
+  pull_request_review:
```

Comment:
Workflow should be started only when PR approved. See `Running a workflow when a pull request is approved` section in documentation